src/simulators/statevector/statevector_state.hpp

```diff
@@ -810,6 +819,15 @@ void State<statevec_t>::measure_reset_update(const std::vector<uint_t> &qubits,
   }
 }
 
+template <class statevec_t>
+void State<statevec_t>::apply_initialize(const reg_t &qubits,
+                                         const cvector_t &params,
+                                         RngEngine &rng) {
+   // Apply reset to qubits
+   apply_reset(qubits, rng);
```

Comment:
Here you could add an optimization to check that if `qubits.size()` is the number of qubits in the state (and they are sorted) just set the vector directly. Eg:

```c++
if (qubits.size() == BaseState::qreg_.num_qubits()) {
  // If qubits is all ordered qubits in the statevector
  // we can just initialize the whole state directly
  auto sorted_qubits = qubits;
  std::sort(sorted_qubits.begin(), sorted_qubits.end());
  if (qubits == sorted_qubits) {
    BaseState::initialize_qreg(qubits.size(), params);
    return;
  }
  // Apply reset to qubits
  apply_reset(qubits, rng);
  // Apply initialize_component
  BaseState::qreg_.initialize_component(qubits, params);
}
```

Reply:
There is no function called "initialize_qreg". Did you mean the following:
`BaseState::qreg_.initialize_from_vector(params);` (see updated PR).

Reply:
Sorry, its just `initialize_qreg` in the current class, not the base state one.